README.md

```diff
@@ -1 +1,52 @@
-# deterministicP256
\ No newline at end of file
+# DeterministicP256 - Deterministically generate P-256 (Secp256r1) credentials
+
+A small library created for [Liquid Auth Android](https://github.com/algorandfoundation/liquid-auth-android) allowing users to specify a BIP39-compliant mnemonic, ran through a PBKDF2, to produce a root seed that can then be used to generate other passkeys for FIDO2.
+
+This allows a user to regenerate passkeys created for specific domains (e.g., GitHub.com) and userIds anew, across platforms and devices. The alternative is to either keep the passkeys device-bound, or to rely on Google or Apple to custody the keys on-behalf of the user.
+
+# Flow
+
+```kotlin
+
+val DP256 = DeterministicP256()
+
+// Generate the root seed by passing along a BIP39 mnemonic:
+
+val rootSeed = DP256.genRootSeedWithBIP39("salon zoo engage submit smile frost later decide wing sight chaos renew lizard rely canal coral scene hobby scare step bus leaf tobacco slice")
+```
+
+The function `genRootSeedWithBIP39(.)` validates the mnemonic, that the words are 1) valid according to the [wordlist](https://github.com/bitcoin/bips/blob/master/bip-0039/bip-0039-wordlists.md). 2) that the checksum (final word) is correct.
+
+It wraps around the private function `genRootSeed` which in turn calls PBKDF2-HMAC-SHA512. By default, the salt is `liquid`(in bytes), iterator count is 600000 and key length 512 (64 bytes).
```

Comment:
according to [owasp](https://cheatsheetseries.owasp.org/cheatsheets/Password_Storage_Cheat_Sheet.html#pbkdf2) when using SHA512 210,000 should do

Reply:
https://support.lastpass.com/s/document-item?language=en_US&_LANG=enus&bundleId=lastpass&topicId=LastPass%2Fabout-password-iterations.html

Last pass has changed their's to 600k

Reply:
Lastpass uses `SHA-256`
I don't think it's critical but it'll probably be good to run with less ops

Reply:
Ultimately the idea is that this is only run once in the lifetime of a device (or until it gets wiped), so if it takes a little longer on a device that's fine. But I'm ok with whatever number.

2048 with the "liquid" pepper might be good enough assuming a good random entropy behind the mnemonic, and 210k/600k is just to protect the brute forcing of the the average user's password (e.g., "password123456") alongside a good salt.

Reply:
Set to 210 000